ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/AbstractEngineNewPayload.java

```diff
@@ -178,6 +180,17 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
               .map(Bytes::fromHexString)
               .map(in -> TransactionDecoder.decodeOpaqueBytes(in, EncodingContext.BLOCK_BODY))
               .collect(Collectors.toList());
+      transactions.forEach(
```

Comment:
is this the same as the code below? or is it more performant to explicitly call the CompletableFuture?
```
      transactions.parallelStream().forEach(
          transaction ->
                    Address sender = transaction.getSender();
                    LOG.atTrace()
                        .setMessage("The sender for transaction {} is calculated : {}")
                        .addArgument(transaction.getHash())
                        .addArgument(sender)
                        .log();
                  );
```

Reply:
Good question, the difference is that parallelStream() will use the main thread as part of the threads that will calculate the transaction senders and hashes, thus the calculation is parallel but not asynchronous. Using CompletableFutures the way it is done in the other code (without join) will make these calculations asynchronous with the thread that is executing the newPayload call.

Reply:
I see, thanks for the explanation.
So a possibly _more asynchronous_ version could be ?
```
CompletableFuture.runAsync(
      transactions.parallelStream().forEach(
          transaction ->
                    Address sender = transaction.getSender();
                    LOG.atTrace()
                        .setMessage("The sender for transaction {} is calculated : {}")
                        .addArgument(transaction.getHash())
                        .addArgument(sender)
                        .log();
                  ));
```

Reply:
Yes, I guess there is not too much difference as iterating over 200 transactions is extremely fast. One other difference is that we can imagine using a completely different thread pool with the first solution if the forkjoin pool is used a lot, by adding predefined executor to the runAsync method.
CompletableFuture.runAsync takes a runnable, so I added () ->  to your suggestion. I thing it is worth testing your suggestion but also the current implementation (in this PR) with an executor.

```
CompletableFuture.runAsync(() -> 
    transactions.parallelStream().forEach(transaction -> {
        Address sender = transaction.getSender();
        LOG.atTrace()
            .setMessage("The sender for transaction {} is calculated : {}")
            .addArgument(transaction.getHash())
            .addArgument(sender)
            .log();
    })
);
```